### ISSUE TITLE
Inconsistency between listContents() and getMetadata()

### ISSUE THREAD

**Issue by systemist** [thephpleague/flysystem#615]
$path = '/path/to/dir/.'; // ends with a dot

$filesystem->getMetadata($path) will return an array with directory metadata;
$filesystem->listContents($path) will return an empty array, as if the directory does not exist or is empty. This is a result of array_filter() call in the Local adapter.

Consider the following scenario, where we try to remove empty directories from a filesystem:

	public function isEmptyDir($dir)
	{
		$contents = $this->filesystem->listContents($dir);

		return sizeof($contents) == 0;
	}

	public function isDir($dir)
	{
		$metadata = $this->filesystem->getMetadata($dir);

		return $metadata['type'] == 'dir';
	}

	public function collapse($dir)
	{
		if (strlen($dir) && $this->isDir($dir) && $this->isEmptyDir($dir))
		{
			$this->filesystem->deleteDir($dir);
			$this->collapse(dirname($dir));
		}
	}

In this case, upon calling collapse($dir), our entire filesystem gets wiped because:
 - there is no way to delete directory unrecursively (unlink doesn't work on dirs)
 - dirname() returns '.' when there is no more parent directory
 - listContents() always returns an empty array when directory ends with a dot

**Comment by frankdejonge** [comment]
@systemist I tried to reproduce this, but:

```php
<?php

use League\Flysystem\Adapter\Local;
use League\Flysystem\Filesystem;

include __DIR__ . '/vendor/autoload.php';

$localAdapter = new Local('/tmp/testing');
$filesystem = new Filesystem($localAdapter);
$filesystem->put('path/to/dir/file.txt', 'lol');

var_dump($filesystem->getMetadata('/path/to/dir/.'));
var_dump($filesystem->listContents('/path/to/dir/.'));
```

yields:

```
array(3) {
  ["type"]=>
  string(3) "dir"
  ["path"]=>
  string(11) "path/to/dir"
  ["timestamp"]=>
  int(1457808125)
}
array(1) {
  [0]=>
  array(8) {
    ["type"]=>
    string(4) "file"
    ["path"]=>
    string(20) "path/to/dir/file.txt"
    ["timestamp"]=>
    int(1457808125)
    ["size"]=>
    int(3)
    ["dirname"]=>
    string(11) "path/to/dir"
    ["basename"]=>
    string(8) "file.txt"
    ["extension"]=>
    string(3) "txt"
    ["filename"]=>
    string(4) "file"
  }
}
```

Can you provide more information? Also, paths like `/path/to/dir/.` will always get normalised and turned into `path/to/dir`. So the `.` should have no influence in this situation.

**Comment by systemist** [comment]
Correct, I assumed a prefixed path to directory, however the prefix is added in constructing Local adapter, so I may have been inaccurate.

Try the following:

	$localAdapter = new Local('/tmp/testing');
	$filesystem = new Filesystem($localAdapter);
	$filesystem->put('path/to/dir/file.txt', 'lol');

	var_dump($filesystem->getMetadata('.'));
	var_dump($filesystem->listContents('.'));

The '.' should be considered a valid path, as it resolves to "/tmp/testing/.". Or not?

**Comment by frankdejonge** [comment]
@systemist correct, a test case has been added, fix has been pushed an version `1.0.19` has been tagged. Should be available on packagist shortly.
